packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.java

```diff
@@ -25,40 +20,51 @@
 import java.io.OutputStream;
 import java.util.Map;
 
-import javax.annotation.Nullable;
+import expo.core.ExportedModule;
+import expo.core.ModuleRegistry;
+import expo.core.Promise;
+import expo.core.interfaces.ActivityEventListener;
+import expo.core.interfaces.ActivityProvider;
+import expo.core.interfaces.ExpoMethod;
+import expo.core.interfaces.ModuleRegistryConsumer;
+import expo.core.interfaces.services.UIManager;
+import expo.core.utilities.FileUtilities;
 
-import host.exp.exponent.ActivityResultListener;
-import host.exp.exponent.Constants;
-import host.exp.exponent.kernel.KernelConstants;
-import host.exp.exponent.utils.ExpFileUtils;
-import host.exp.exponent.utils.ScopedContext;
-import host.exp.expoview.Exponent;
+public class DocumentPickerModule extends ExportedModule implements ModuleRegistryConsumer, ActivityEventListener {
 
+  private static final String TAG = "ExpoDocumentPicker";
 
-public class DocumentPickerModule extends ReactContextBaseJavaModule implements ActivityResultListener {
   private static int OPEN_DOCUMENT_CODE = 4137;
+  private ModuleRegistry mModuleRegistry;
+  private ActivityProvider mActivityProvider;
+  private UIManager mUIManager;
 
-  private @Nullable Promise mPromise;
-  private ScopedContext mScopedContext;
-
+  @Nullable
+  private Promise mPromise;
   private boolean mCopyToCacheDirectory = true;
 
-  public DocumentPickerModule(ReactApplicationContext reactContext, ScopedContext scopedContext) {
-    super(reactContext);
-    mScopedContext = scopedContext;
-
-    Exponent.getInstance().addActivityResultListener(this);
+  public DocumentPickerModule(Context context) {
+    super(context);
   }
 
   @Override
   public String getName() {
-    return "ExponentDocumentPicker";
+    return TAG;
+  }
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    mModuleRegistry = moduleRegistry;
```

Comment:
It's a good practice to unregister (and check if `moduleRegistry != null` when calling methods on it)

Reply:
I've checked the codebase and there's no possible way to `unregister` `activityResultListener`